api/src/test/java/org/apache/iceberg/types/TestTypeUtil.java

```diff
@@ -450,20 +449,14 @@ public void testProjectListNested() {
                                     required(17, "x", Types.IntegerType.get()),
                                     required(18, "y", Types.IntegerType.get()))))))));
 
-    AssertHelpers.assertThrows(
-        "Cannot explicitly project List",
-        IllegalArgumentException.class,
-        () -> TypeUtil.project(schema, Sets.newHashSet(12)));
+    Assertions.assertThatThrownBy(() -> TypeUtil.project(schema, Sets.newHashSet(12)))
+        .isInstanceOf(IllegalArgumentException.class);
```

Comment:
these should all check for a specific error msg, as otherwise the called code could fail for a different reason that is being expected